src/librustc_mir/interpret/step.rs

```diff
@@ -117,6 +117,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
             // Statements we do not track.
             AscribeUserType(..) => {}
+            Coverage(..) => {}
```

Comment:
This makes that statement a NOP for Miri/CTFE. So it is always correct to just do nothing here, and there is no UB that can possibly be caused by these statements?

Reply:
Thanks for the callout. Good question.

The referenced line of code is a great place for me to re-insert the FIXME comment I had removed from rustc:mir::interpret::intrinsics. I just [pushed a change adding some comments here, for `step`](https://github.com/rust-lang/rust/pull/75563/files#diff-fabaa9592cbf2a32d5acb6e8cc047908).

Correct me if I'm wrong, but I believe it is reasonable to treat the Coverage statement as a NOP, if evaluated in a const expression, at this time. The statement is injected by a MIR pass, in a way that should not have any side effects on the runtime behavior, so ignoring it here effectively "un-injects" it, as if it never existed (right?).

If the Coverage statement is a "Counter" kind, it would normally result in LLVM-generated runtime code of only three statements: load a global int, increment it, and store it. 

If and/or what we might want to do in the future is the subject of the FIXME issue #73156.

Reply:
> Correct me if I'm wrong, but I believe it is reasonable to treat the Coverage statement as a NOP, if evaluated in a const expression, at this time. The statement is injected by a MIR pass, in a way that should not have any side effects on the runtime behavior, so ignoring it here effectively "un-injects" it, as if it never existed (right?).

Yes, that sounds right.

Note that your comment only seems to take into account CTFE. This code here is also used by [the Miri UB checker](https://github.com/rust-lang/miri/). So it has to adequately model all UB. But since coverage does not exist on the source level, and since presumably it is not something the program can interact with directly, I assume there also is no UB to check for here.

The comment should be adjusted taking into account that this is not just for const expressions. This is for all code that is evaluated with the Miri engine.

Reply:
Got it. I'll make that change. Thanks.

Reply:
done

Reply:
Looking good, thanks. :)